### PR TITLE
Replace actions-rs GitHub actions

### PR DESCRIPTION
**Pull request by saschagrunert**
Those actions are not maintained anymore and we should switch to an alternative.

We also remove the `-Cinline-threshold=0` rustflag from the unit tests
because that's now a noop.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -57,12 +57,9 @@ jobs:
             target
           key: ${{ runner.os }}-cargo-clippy-${{ hashFiles('**/Cargo.lock') }}
       - name: Select Toolchain
-        uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # v1.0.7
+        uses: dtolnay/rust-toolchain@7b1c307e0dcbda6122208f10795a713336a9b35a
```
Comment (flavio) [inline .github/workflows/ci.yml:60]: Would you mind adding the "human" version next to the shasum?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -80,31 +78,21 @@ jobs:
...
-        uses: actions-rs/cargo@844f36862e911db73fe0815f00a4a2602c279505 # v1.0.3
-        with:
-          command: test
-          args: --all-features --no-fail-fast
+        run: cargo test --all-features --no-fail-fast
         env:
           CARGO_INCREMENTAL: '0'
-          RUSTFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Zpanic_abort_tests'
-          RUSTDOCFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Zpanic_abort_tests'
+          RUSTFLAGS: '-Zprofile -Ccodegen-units=1 -Clink-dead-code -Coverflow-checks=off -Zpanic_abort_tests'
```
Comment (flavio) [inline .github/workflows/ci.yml:88]: nit, the change the the `RUSTFLAGS` is not related with the GH action switch. Can you put that into another commit or add some explanation to the current commit message?

**Review by flavio — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (saschagrunert) [reply .github/workflows/ci.yml:60]: There is none unfortunately, I opened https://github.com/dtolnay/rust-toolchain/issues/121 because of that.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (saschagrunert) [reply .github/workflows/ci.yml:88]: Sure, I'll update the commit message accordingly.

**Review by cgwalters — APPROVED** [review]

**Comment by saschagrunert** [comment]
@flavio merging is blocked:

> [!CAUTION]
> Merging can be performed automatically once the requested changes are addressed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (flavio) [reply .github/workflows/ci.yml:60]: I've seen the response on the linked issue. There's nothing we can do about that

**Review by flavio — APPROVED** [review]

**Comment by flavio** [comment]
@saschagrunert :rocket:
